Formula/gnupg.rb

```diff
@@ -13,53 +13,83 @@ class Gnupg < Formula
+
+  deprecated_option "without-libusb-compat" => "without-libusb"
+
+  depends_on "pkg-config" => :build
+  depends_on "sqlite" => :build if MacOS.version == :mavericks
+  depends_on "npth"
+  depends_on "gnutls"
+  depends_on "libgpg-error"
+  depends_on "libgcrypt"
+  depends_on "libksba"
+  depends_on "libassuan"
+  depends_on "pinentry"
+  depends_on "gettext"
+  depends_on "adns"
+  depends_on "libusb" => :recommended
+  depends_on "readline" => :optional
+  depends_on "homebrew/fuse/encfs" => :optional
+
+  # Remove for > 2.1.19
+  # ssh-import.scm fails during "make check" for sandboxed builds
+  # Reported 1 Mar 2017 https://bugs.gnupg.org/gnupg/issue2980
+  patch do
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/6d87027/gnupg/agent-handle-scdaemon-errors.patch"
+    sha256 "c2de625327c14b2c0538f45749f46c67169598c879dc81cf828bbb30677b7e59"
+  end
 
   def install
     args = %W[
       --disable-dependency-tracking
       --disable-silent-rules
       --prefix=#{prefix}
-      --disable-asm
-      --program-suffix=1
+      --sbindir=#{bin}
+      --sysconfdir=#{etc}
+      --enable-symcryptrun
+      --with-pinentry-pgm=#{Formula["pinentry"].opt_bin}/pinentry
     ]
-    args << "--with-libusb=no" if build.without? "libusb-compat"
+
+    args << "--disable-ccid-driver" if build.without? "libusb"
+    args << "--with-readline=#{Formula["readline"].opt_prefix}" if build.with? "readline"
 
     system "./configure", *args
-    system "make"
-    system "make", "check"
 
-    # we need to create these directories because the install target has the
-    # dependency order wrong
-    [bin, libexec/"gnupg"].each(&:mkpath)
     system "make", "install"
 
-    # https://lists.gnupg.org/pipermail/gnupg-devel/2016-August/031533.html
-    inreplace bin/"gpg-zip1", "GPG=gpg", "GPG=gpg1"
+    # "make check" cannot run before "make install"
+    # Reported 1 Mar 2017 https://bugs.gnupg.org/gnupg/issue2979
+    system "make", "check"
+
+    # Add symlinks from gpg2 to unversioned executables, replacing gpg 1.x.
+    bin.install_symlink "gpg2" => "gpg"
+    bin.install_symlink "gpgv2" => "gpgv"
+    man1.install_symlink "gpg2.1" => "gpg.1"
+    man1.install_symlink "gpgv2.1" => "gpgv.1"
 
-    # Although gpg2 support should be pretty universal these days
-    # keep vanilla `gpg` executables available, at least for now.
-    %w[gpg-zip1 gpg1 gpgsplit1 gpgv1].each do |cmd|
-      (libexec/"gpgbin").install_symlink bin/cmd => cmd.to_s.sub(/1/, "")
-    end
+    bin.install "tools/gpgsplit" => "gpgsplit2" if build.with? "gpgsplit"
   end
 
-  def caveats; <<-EOS.undent
-    This formula does not install either `gpg` or `gpgv` executables into
-    the PATH.
-
-    If you simply require `gpg` and `gpgv` executables without explicitly
-    needing GnuPG 1.x we recommend:
-      brew install gnupg2
+  def post_install
+    (var/"run").mkpath
+    Utils.popen_read("killall", "gpg-agent")
```

Comment:
Note that as written, this prints nothing. To get the ugly message I'd need to do
```
puts Utils.popen_read("killall gpg-agent 2>&1")
```